src/domains/wallet-app/api/npg-notifications/v1/_base_policy.xml.tpl

```diff
@@ -51,11 +51,19 @@
                 details = new JObject();
                 details["type"] = "PAYPAL";
                 details["maskedEmail"] = (string)operation["paymentInstrumentInfo"];
+            } else if(paymentCircuit == "CARD"){
+                details = new JObject();
+                details["type"] = "CARD";
+                if(additionalData.Type != JTokenType.Null){
+                    details["paymentInstrumentGatewayId"] = (string)((JObject)additionalData)["cardId4"];
```

Comment:
maybe you can move up check performed in line 62 above and avoid repeating the `if(additionalData.Type != JTokenType.Null){` multiple times, assigning a local variables to be used here. What do you think?